nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thomas Roger
+ */
+
+package org.nuxeo.wopi;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.common.Environment;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.model.Property;
+import org.nuxeo.ecm.core.utils.BlobsExtractor;
+import org.nuxeo.runtime.model.ComponentContext;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/**
+ * @since 10.3
+ */
+public class WOPIServiceImpl extends DefaultComponent implements WOPIService {
+
+    private static final Logger log = LogManager.getLogger();
+
+    public static final String WOPI_DIR = "wopi";
+
+    public static final String DISCOVERY_XML = "discovery.xml";
+
+    protected static final BlobsExtractor BLOBS_EXTRACTOR = new BlobsExtractor();
+
+    // extension => app name
+    protected Map<String, String> extensionAppNames = new HashMap<>();
+
+    // extension => wopi action => wopi action url
+    protected Map<String, Map<String, String>> extensionActionURLs = new HashMap<>();
+
+    @Override
+    public void start(ComponentContext context) {
+        File wopiDir = new File(Environment.getDefault().getData(), WOPI_DIR);
+        File discoveryFile = new File(wopiDir, DISCOVERY_XML);
+        if (!discoveryFile.exists() || !discoveryFile.isFile()) {
+            log.error("Cannot load discovery file at: {}, WOPI disabled.", discoveryFile.getAbsolutePath());
```

Comment:
Don't remember if it's a good idea to log at error level the absolute path (security wise)?

Reply:
I don't think so, furthermore as the location is static I don't see any blocker to just log something without the path.